packages/components/src/internal/components/samples/utils.tsx

```diff
@@ -297,3 +302,29 @@ export function getStorageItemUpdateData(
     };
 }
 
+export function getSampleStatusLockedMessage(state: SampleState, saving: boolean) : string {
+    let msgs = [];
+    if (state?.inUse || saving)
+        msgs.push('cannot change status type or be deleted because it is in use');
+    if (state && !state.isLocal)
+        msgs.push('can be changed only in the ' + state.containerPath.substring(1) + ' project');
+    if (msgs.length > 0)
+        return 'This sample status ' + msgs.join(' and ') + '.'
+    return undefined;
+}
+
+export function getSampleStatusContainerFilter(forLegend?: boolean, containerPath?: string, moduleContext?: ModuleContext) : Query.ContainerFilter {
```

Comment:
This is great that you put this into a helper function instead of just having container filters dispersed in the code for the various scenario!